pkg/pool-utils/contracts/controllers/AssetManagedLBPController.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/ERC20Helpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+
+import "../interfaces/IControlledLiquidityBootstrappingPool.sol";
+import "./BasePoolController.sol";
+
+/**
+ * @notice Controller for an unseeded LiquidityBootstrappingPool. It also serves as the asset manager.
+ * Using the `UnseededLiquidityBootstrappingPoolFactory` will deploy this controller and a pool.
+ * After the contracts are deployed by the factory, the manager can call `fundPool` on the controller
+ * with the initial balances. This funds the pool using the manager's pool tokens. The reserve tokens
+ * will be supplied using an AAVE flashloan, then immediately withdrawn using the asset manager
+ * functionality to repay the loan.
+ *
+ * The result is an apparently fully funded pool, but with a zero cash balance in the reserve token.
+ * This means project tokens cannot be "sold back" into the pool - at least until enough are sold to
+ * build up the real balance.
+ *
+ * Note that there is a flash loan fee, so the manager will need enough reserve tokens to pay it (and
+ * must approve the controller to pull them during the `fundPool` call). The amount of the fee can be
+ * calculated using the `getFlashLoanFeePercentage` function on the controller.
```

Comment:
The current version doesn't need a flash loan (though it would be possible to do it that way): I updated the comment. There shouldn't be any references to flash loans in the code. Is this from an older commit?

Reply:
Charging a fee for this service could also be a good idea to filter out non reputable projects